### PR TITLE
Issue #5452: Avoid Wercker failures in not updated PRs after release

### PR DESCRIPTION
**Pull request by Vladlis**
#5452

depends on https://github.com/checkstyle/contribution/pull/277

### CODE REVIEW

**Comment by romani** [comment]
Wercker build is weird and miss bunch of build items, please rebase on most latest master.

**Comment by Vladlis** [comment]
@romani 
This is done for a reason: parent commit has version 8.8, while current version is 8.9-SNAPSHOT, also I've removed "no-error" jobs to speed up testing, as we only need to test "no-exception" jobs now

**Comment by romani** [comment]
@Vladlis ,

to test you fix in contribution you need to have changes ONLY (in PR):
```
for i in 1 2 3 4 5; do git clone https://github.com/vladlis/contribution && break || sleep 15; done
git checkout -b 276-cs-version
```
no other changes are expected.
Your current substitution to version `8.8` is not clear. During build we operate only by SNAPSHOT versions.

BUT you have to reset your forked master branch to any commit commit before release to have in POM file 8.7-SNAPSHOT version. Do some fake changes in any file, and send PR.
As CI pass I should not merge any changes to main repo and just close this PR.

**Comment by romani** [comment]
@Vladlis , is there any chance for you to finish ?
if you have problems, please let me know.

**Comment by Vladlis** [comment]
@romani Sorry for delay, I will continue after 11th March. If you consider this a high priority task please feel free to re-assign the issue.

**Comment by romani** [comment]
Please continue

**Comment by romani** [comment]
@Vladlis , is there any chance to finish this ?

**Comment by romani** [comment]
@Vladlis , please rebase, contribution PR is merged.

**Comment by Vladlis** [comment]
@romani, rebased
also test PR is created: https://github.com/checkstyle/checkstyle/pull/5649
and looks like testing is successful as Apache Struts no-exception test passes even though `CS_version: 8.9-SNAPSHOT` in branch while master version is 8.8.1-SNAPSHOT

**Comment by romani** [comment]
Yes

**Comment by rnveach** [comment]
circle error:
````
[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.3.1:exec (default-cli): Goal requires a project to execute but there is no POM in this directory (/home/ubuntu). Please verify you invoked Maven from the correct directory. -> [Help 1] org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.3.1:exec (default-cli): Goal requires a project to execute but there is no POM in this directory (/home/ubuntu). Please verify you invoked Maven from the correct directory.
````

**Comment by Vladlis** [comment]
@romani @rnveach 
I feel like I am spending too much time on figuring out of how to make this work on Travis and CircleCI while this should be easy, so any help would be appreciated.

**Comment by romani** [comment]
travis passed. Shippable is relaunched ... should pas too.

**Comment by Vladlis** [comment]
@romani @rnveach 
Please take a look now

File: circle.yml [hunk 1]
```diff
@@ -26,7 +26,7 @@ machine:
     DEP3: " && /home/ubuntu/checkstyle/groovy-2.4.7/bin/groovy launch.groovy --listOfProjects projects-to-test-on.properties --config my_check.xml"
     TESTER_DEPENDENCIES: $CMD1$DEP1$DEP2$DEP3
     CMD2: " && sed -i'' 's/^openjdk/#openjdk/' projects-for-circle.properties"
-    CMD3: " && /home/ubuntu/checkstyle/groovy-2.4.7/bin/groovy launch.groovy --listOfProjects projects-for-circle.properties --config checks-nonjavadoc-error.xml"
+    CMD3: " && cd ../.. /home/ubuntu/checkstyle/groovy-2.4.7/bin/groovy contribution/checkstyle-tester/launch.groovy --listOfProjects projects-for-circle.properties --config checks-nonjavadoc-error.xml --checkstyleVersion ${CS_POM_VERSION}"
```
Comment (romani) [inline circle.yml]: Please explain reason of this "cd" .
the less "cd" is used in shell script, the better.

Comment (rnveach) [reply circle.yml]: please also put `&&` between `cd` command ` groovy` command so it is clear where the `cd` command stops and and the `groovy` command begins. I mistakenly thought you were cding into a groovy directory.

Comment (Vladlis) [reply circle.yml]: @rnveach done
@romani I was only able to make it work this way. Without `cd` I was getting 
> Goal requires a project to execute but there is no POM in this directory (/home/ubuntu).

like here: https://circleci.com/gh/Vladlis/checkstyle/324

Comment (Vladlis) [reply circle.yml]: well, CI was green only due to a missed `cd` and now this doesn't work either

**Pull request merged**